test/others/test_utils.py

```diff
@@ -1193,3 +1198,26 @@ def test_get_eval_run_results():
     first_result = node_results.iloc[0]
     assert first_result["exact_match"] == True
     assert first_result["answer"] == "This"
+
+
+def test_exponential_backoff():
+    # Test that the exponential backoff works as expected
+    # should raise exception, check the exception contains the correct message
+    with pytest.raises(Exception, match="retries \(2\)"):
+
+        @retry_with_exponential_backoff(backoff_in_seconds=1, max_retries=2)
+        def greet(name: str):
+            if random() < 0.999:
```

Comment:
I think to make this safer could we just make this always returns a `raise OpenAIRateLimitError` and remove the if statement? Or make the if statement always fail, e.g. `if random() < 1.1:`. This way we would avoid having this test occasionally fail when `random()` happens to return `1.0`.